packages/steps/src/BuildStep.ts

```diff
@@ -281,6 +288,35 @@ export class BuildStep extends BuildStepOutputAccessor {
     );
   }
 
+  public shouldExecuteStep(hasAnyPreviousStepsFailed: boolean): boolean {
+    const defaultStatusCheck = 'success';
+    const statusCheck = this.ifCondition
+      ? getSelectedStatusCheckFromIfStatementTemplate(this.ifCondition)
+      : defaultStatusCheck;
+
+    switch (statusCheck) {
+      case 'success':
+        return !hasAnyPreviousStepsFailed;
+      case 'failure':
+        return hasAnyPreviousStepsFailed;
+      case 'always':
+        return true;
+    }
+  }
+
+  public skip(): void {
+    this.status = BuildStepStatus.SKIPPED;
+    this.ctx.logger.info(
+      { marker: BuildStepLogMarker.START_STEP },
+      'Executing build step "${this.displayName}"'
+    );
+    this.ctx.logger.info(`Skipped build step "${this.displayName}"`);
+    this.ctx.logger.info(
+      { marker: BuildStepLogMarker.END_STEP, result: BuildStepStatus.SKIPPED },
+      `Skipped build step "${this.displayName}"`
+    );
```

Comment:
We need those three logs to start and end step phase? My first reaction was "why not one `.info()`".

Reply:
Yeah unfortunately we do, so it is displayed nicely on the website. I also tried with just one line, but it wasn't working.

BTW only `Skipped build step "${this.displayName}"` will be visible in the logs